test/FunctionalTests/CompressionTests.cs

```diff
@@ -0,0 +1,311 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FunctionalTestsWebsite.Services;
+using Greet;
+using Grpc.AspNetCore.FunctionalTests.Infrastructure;
+using Grpc.AspNetCore.Server.Compression;
+using Grpc.AspNetCore.Server.Internal;
+using Grpc.AspNetCore.Server.Tests;
+using Grpc.Core;
+using NUnit.Framework;
+
+namespace Grpc.AspNetCore.FunctionalTests
+{
+    [TestFixture]
+    public class CompressionTests : FunctionalTestBase
```

Comment:
it might be a good idea to try to add test scenarios that match the behavior described in the compression interop tests:
https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#server_compressed_unary
https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#server_compressed_streaming

Reply:
That is covered with this functional test: SendCompressedMessageAndReturnResultWithNoCompressFlag_ResponseNotCompressed